walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,6 +272,151 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
-  
+
+  @Test
+  public void optionalAttendeeNotConsidered() {
+    // Have each person have different events. We should see three options because each person has
+    // split the restricted times. There is an optional attendee and their event isn't considered.
+    //
+    // OptEvent: |-------------C---------------|
+    // Events  :       |--A--|     |--B--|     
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY,
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalAttendeeConsidered() {
+      // Have each person have different events. We should see three options because each person has
+    // split the restricted times. There is an optional attendee and their event is considered.
```

Comment:
The previous test doesn't consider the optional attendee, but this test does. I don't obviously see where that option is set (is it a flag? or something else that triggers it?).

Reply:
It seems that the optional attendee's schedule is considered if their event has the same TimeRange as an available meeting time. It's kind of an odd requirement but that's why the optional attendee is considered in the test.

Reply:
I think I just don't see where in the code the optional attendees are considered at all.

Reply:
I don't have the optional feature in my code because the walk through stated that if there is extra time I could add code for the optional tests. Unfortunately I was behind, so I couldn't prioritize this section.

Reply:
ok, sgtm.